sdk/servicebus/service-bus/test/internal/node/operationOptionsForAMQP.spec.ts

```diff
@@ -0,0 +1,985 @@
+        });
+
+        after(async () => {
+          await sbAdminClient.deleteQueue(queueName);
+        });
+
+        it("RequestOptions is plumbed through acceptSession", async () => {
+          await verifyOperationOptionsAtGetToken(
+            sbClient["_connectionContext"],
+            sampleOperationOptions,
+            async () => {
+              await sbClient.acceptSession("queue", "session-id");
+            }
+          );
+        });
+
+        it("RequestOptions is plumbed through acceptNextSession", async () => {
+          await verifyOperationOptionsAtGetToken(
+            sbClient["_connectionContext"],
+            sampleOperationOptions,
+            async () => {
+              await sbClient.acceptNextSession("queue");
+            }
+          );
+        });
+
+        it("RequestOptions is plumbed through renewSessionLock", async () => {
+          const receiver = await sbClient.acceptNextSession(queueName);
+          await verifyOperationOptionsAtGetToken(
+            sbClient["_connectionContext"],
+            sampleOperationOptions,
+            async () => {
+              await receiver.renewSessionLock();
+            }
+          );
+        });
+
+        it("RequestOptions is plumbed through setSessionState", async () => {
+          const receiver = await sbClient.acceptNextSession(queueName);
+          await verifyOperationOptionsAtGetToken(
+            sbClient["_connectionContext"],
+            sampleOperationOptions,
+            async () => {
+              await receiver.setSessionState("set-state");
+            }
+          );
+        });
+
+        it("RequestOptions is plumbed through getSessionState", async () => {
+          const receiver = await sbClient.acceptNextSession(queueName);
+          await verifyOperationOptionsAtGetToken(
+            sbClient["_connectionContext"],
+            sampleOperationOptions,
+            async () => {
+              await receiver.getSessionState();
+            }
+          );
+        });
+
+        it("RequestOptions is plumbed through peekMessages", async () => {
+          const receiver = await sbClient.acceptNextSession(queueName);
+          await verifyOperationOptionsAtGetToken(
+            sbClient["_connectionContext"],
+            sampleOperationOptions,
+            async () => {
+              await receiver.peekMessages(1);
+            }
+          );
+        });
+
+        it("RequestOptions is plumbed through receive deferred messages", async () => {
+          const receiver = await sbClient.acceptNextSession(queueName);
+          await verifyOperationOptionsAtGetToken(
+            sbClient["_connectionContext"],
+            sampleOperationOptions,
+            async () => {
+              await receiver.receiveDeferredMessages(Long.ZERO);
+            }
+          );
+        });
+      });
+    });
+  });
+
+  describe("AbortSignal", () => {
+    // TODO: Add tests to make sure abortSignal works at each level
```

Comment:
Really you just need to make sure it gets propagated. No need to test all behavior.

Reply:
Hmmm, now that I think about it, you're actually right. I was initially wondering that complicated methods such as `receiveMessages` consume and manage abortSignal internally rather than just passing around. I meant this test-suite for such cases.

But, those complicated methods already used to allow passing `abortSignal` and have the tests. All the new async methods that have the abortSignal now just pass it around, so I can merge this with the requestOptions test suite above.